### ISSUE TITLE
Exiting symbol editing mode puts Emacs in Vim style editing

### ISSUE THREAD

**Issue by shivamkalra** [syl20bnr/spacemacs#5060]
#### Description
Existing symbol editing mode puts Emacs in Vim style mode. 

#### Reproduction guide
- Start Emacs
- Press `M-m s e` to enter symbol editing mode
- Press `e` to make an edit
- Do editing and exiting symbol editing mode by pressing escape.

*Observed behaviour:*
It puts Emacs in Vim mode and cannot switch back to Emacs style editing.

*Expected behaviour:*
It should come back to whichever editing style user was using before.

![result](https://cloud.githubusercontent.com/assets/1386352/13030950/fed51fc4-d288-11e5-94c7-399f6d1658b8.gif)

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.9
- Spacemacs branch: develop (rev. 9f862c8)
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-enable-help-tooltip t)
 better-defaults spacemacs-helm emacs-lisp gtags c-c   version-control git markdown org gnus shell syntax-checking csharp sql csharp lua elpy xkcd haskell extra-langs emoji spell-checking ipython-notebook
 (erc :variables erc-sasl-server-regexp-list
      '("irc\.freenode\.net"))
 shell-scripts theming javascript html colors ibuffer smex github themes-megapack fasd wakatime notmuch mu4e ipython-notebook matlab search-engine tools
 (elfeed :variables rmh-elfeed-org-files
         (list "~/.emacs.d/private/elfeed.org"))
 (latex :variables latex-enable-auto-fill t))
```

**Comment by StreakyCobra** [comment]
I think it is expected right now because `M-m s e` is bound to `evil-iedit-state` which is meant to work with `evil`. Can you try with [`C-;`](https://github.com/victorhge/iedit/blob/2d9017daeaeb408568992f5c9cad5df5ca4f1d12/iedit.el#L97) instead?

**Comment by shivamkalra** [comment]
@StreakyCobra Thanks. Yes, `C-;` works way I would expect it to. However, wouldn't it be better if `M-m s e` could decide this based on `dotspacemacs-edit-style` variable?

```
(if (eq dotspacemacs-editing-style 'emacs)
           (iedit-mode)
         (evil-iedit-state))
```

**Comment by StreakyCobra** [comment]
It seems it would be better yes. But I am not really used to the emacs editing style, so I'm not sure. What you can do is create a PR with, it will be discussed there :-)

**Comment by AlejandroCatalina** [comment]
Actually, we can either disable the command under emacs editing style (as holy-mode has `C-;`) or modify that function.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
